### ISSUE TITLE
Transaction chapter needs to updated to match wsittutorial.zip

### ISSUE THREAD

**Issue by glassfishrobot** [javaee/metro-wsit#509]
----------------------------------
Transaction Chapter review status:
----------------------------------
The Transactions Chapter in M4/WSITTutorial.pdf & M4/index.html has not been
updated yet to be in sync with the latest Tx example bits contained in new
wsittutorial.zip. I have reviewed on Solaris the new and improved Tx example
taken from wsittutorial.zip, and I've found a few issues/bugs as shown below.
Once these bugs are fixed, they'll have to merge the new Tx example instructions
contained in wsittutorial.zip into Transactions Chapter in M4/WSITTutorial.pdf &
M4/index.html, and then M4/WSITTutorial.pdf & M4/index.html will need to be
reviewed again.

So long as my comments/bug fixes from his standalone version of Tx example
included in wsittutorial.zip are incorporated back into the real
wsittutorial.zip that we are shipping, and the text of Transactions chapter in
M4/WSITTutorial.pdf & M4/index.html gets updated correctly from new
wsittutorial.zip file
<wsit.tutorial.home>/wsittutorial/examples/wstx/basicWSTX/readme.html than there
should be no deltas between what I've already reviewed in this standalone
version of new wsittutorial.zip and what is contained in the released bits.

Issues testing on Solaris, following
<wsit.tutorial.home>/wsittutorial/examples/wstx/basicWSTX/readme.html contained
in new wsittutorial.zip:

1). wsittutorial/examples/wstx/basicWSTX/SampleService/build.xml

change the slashes to $

{path.separator}

or try "/" instead of "\" for

<param name="target.domain" value="$

{javaee.home}\domains\domain2\config" />

after changing target.domain to use unix slash, this "trust-domains" worked for
me on solaris:

<target name="trust-domains" depends="tools" description="enable domain1 and
domain2 to trust each other by installing each doma
ins s1as security certificate in other domains truststore">
<antcall target="trust-other-domain">
<param name="source.domain"
value="${javaee.home}

\domains\domain1\config" />
<param name="target.domain"
value="$

{javaee.home}/domains/domain2/config" />
<param name="source.alias" value="s1as"/>
<param name="target.alias" value="domain1_s1as"/>
</antcall>
<echo message=" Domain1 self-signed security certificate, alias s1as,
added to domain2 truststore as alias domain1_s1as"/>

<exec executable="${asadmin}" failonerror="false">
<arg line=" stop-domain domain2"/>
</exec>
<antcall target="trust-other-domain">
<param name="source.domain"
value="${javaee.home}

\domains\domain2\config" />
<param name="target.domain"
value="$

{javaee.home}/domains/domain1/config" />
<param name="source.alias" value="s1as"/>
<param name="target.alias" value="domain2_s1as"/>
</antcall>
<echo message=" Domain2 self-signed security certificate, alias s1as,
added to domain1 truststore as alias domain2_s1as"/>
<exec executable="${asadmin}" failonerror="false">
<arg line=" stop-domain domain1"/>
</exec>
</target>

2). wsittutorial/examples/wstx/basicWSTX/SampleService/build.xml

this ant target fails, because the domain1 server is not running, need to ensure
the domain1 is running before invoking this ant target:

% ant create-queue
Buildfile: build.xml

create-queue:
Duplicated project name in import. Project SampleService-impl defined first in
/wsittutorial/examples/wstx/basicWSTX/SampleService/nbproject/ant-deploy.xml and
again in
/wsittutorial/examples/wstx/basicWSTX/SampleService/nbproject/build-impl.xml

check:

tools:

create-jms-resource:
[exec] Unable to connect to admin-server at given host: [localhost] and
port: [4848].
[exec] Please check if this server is up and running and that the host and
port provided are correct.
[exec] CLI137 Command create-jms-resource failed.
[exec] Result: 1

3). need to change how you start domain1 so that it actually is running after
ant target invocation. i changed the start of domain1 to look like this "jon"
debug only ant target and it works, domain1 actually gets started, so now
create-queue will work since server is running:

<target name="jon" depends="tools">
<echo message=" Starting ${javaee.home}

domain1 so domain2 will be
created with ports that do not conflict with domain1"/>
<exec executable="$

{asadmin}" failonerror="true">
<arg line="start-domain"/>
<arg line="--user admin"/>
<arg line="--passwordfile=${javaee.server.passwordfile}"/>
<arg line="domain1"/>
</exec>

<waitfor maxwait="2" maxwaitunit="minute">
<http url="http://${javaee.server.name}:${javaee.server.port}"/>
</waitfor>
<echo message="domain1 started on
[http://$](http://$){javaee.server.name}:${javaee.server.port}."/>
</target>

% ant jon
Buildfile: build.xml

check:

tools:

jon:
[echo] Starting
/net/galapago.east/files/share/workspaces/m4_wsit_tutorial/glassfish domain1 so
domain2 will be created with ports that do not conflict with domain1
[exec] Starting Domain domain1, please wait.
[exec] Log redirected to
/net/galapago.east/files/share/workspaces/m4_wsit_tutorial/glassfish/domains/domain1/logs/server.log.
[exec] Redirecting output to
/net/galapago.east/files/share/workspaces/m4_wsit_tutorial/glassfish/domains/domain1/logs/server.log
[exec] Domain domain1 is ready to receive client requests. Additional
services are being started in background.
[exec] Domain [domain1] is running [Sun Java System Application Server 9.1
(build b41-beta2)] with its configuration and logs at:
[/net/galapago.east/files/share/workspaces/m4_wsit_tutorial/glassfish/domains].
[exec] Admin Console is available at [http://localhost:4848](http://localhost:4848).
[exec] Use the same port [4848] for "asadmin" commands.
[exec] User web applications are available at these URLs:
[exec] [http://localhost:8080 https://localhost:8181](http://localhost:8080 https://localhost:8181).
[exec] Following web-contexts are available:
[exec] [/web1 /__wstx-services ].
[exec] Standard JMX Clients (like JConsole) can connect to JMXServiceURL:
[exec] [service:jmx:rmi:///jndi/rmi://galapago.east.sun.com:8686/jmxrmi]
for domain management purposes.
[exec] Domain listens on at least following ports for connections:
[exec] [8080 8181 4848 3700 3820 3920 8686 ].
[exec] Domain does not support application server clusters and other
standalone instances.

[echo] domain1 started on [http://localhost:8080](http://localhost:8080).

BUILD SUCCESSFUL

so replace

<exec executable="${asadmin}" spawn="true">
<arg line=" start-domain --user admin --passwordfile
${wsit.tutorial.home}

/examples/bp-project/admin-password.txt"/>
</exec>

with

<exec executable="$

{asadmin}

" failonerror="true">
<arg line="start-domain"/>
<arg line="--user admin"/>
<arg line="--passwordfile=$

{javaee.server.passwordfile}

"/>
<arg line="domain1"/>
</exec>

in create-domain2 target.

4). wsittutorial/examples/wstx/basicWSTX/readme.html

6 d.

Select the WSIT tab and open the Operation node in each of the following
sections and select the indicated setting from the Transaction pulldown:

*   init to Required
*   publishRequired to Required
*   publishSupports to Supported
*   verify to Required

the tab is now called "WSIT Configuration" and there is only 1 tab, so you don't
have to select it from anything, since there is no other tabs to choose from.

5). wsittutorial/examples/wstx/basicWSTX/readme.html

6 d.

Select the WSIT tab and open the Operation node in each of the following
sections and select the indicated setting from the Transaction pulldown:

*   init to Required
*   publishRequired to Required
*   publishSupports to Supported
*   verify to Required

under the "Wsit Configuration" tab, there are two nodes titled
"SimpleServiceBindingBinding Binding", why? and which one should i choose to
expand in order to perform the above step mentioned in 6d.?

6). step 9 from wsittutorial/examples/wstx/basicWSTX/readme.html won't let me
click finish after entering package name wstx.sample.client

here is the text i'm following:

Create web service references for the client (two web service clients, a
simpleServlet and a CMT EJB client) and generate the WSDL for both.

1.

In the Projects tab, right-click SampleServiceClient, select New, then
select Web Service Client.
2.

Click Browse next to the Project field. The Browse Web Services dialog is
displayed.
3.

Open SampleService-war, select Simple, then click OK.
4.

In the Package field, enter wstx.sample.client, then click Finish.

i can't perform this step as New Web Service Client wizard states in red letters

"X Package already used for holding SimpleService[client] artifacts."

and the Next and Finish buttons are greyed out.

i'm blocked, so what is the correct package name in order to proceed without
errors here?

7). same is true for steps 9 e. thought 9 f. from
wsittutorial/examples/wstx/basicWSTX/readme.html

#

Repeat Steps 9a and 9b.
#

Open SampleService-ejb, select SimpleASCMTEjb, then click OK.
#

In the Package field, enter wstx.sample.ejbclient, then click Finish.

if i enter package name wstx.sample.ejbclient

i see this in red letters:

"X Package already used for holding SimpleASCMTEjb[client] artifacts."

and the Next and Finish buttons are greyed out.

i'm blocked again...

do we need to do step 9, since SampleServiceClient-->Web Service References is
already populated with two nodes "SimpleAsCMTEjb" and "SimpleService" so we'd
be creating two more references named "SimpleAsCMTEjb_1" and "SimpleService_1"

or should these two pre-configured Web Service References not be present when
first expanding SampleServiceClient-->Web Service References node, and still ask
users to do step 9?

8). now when i delete the "SimpleService" Web Service
References node, and try to recreate it by following step 9, i get an error:

Web Service Client can not be created by JAXWS:wsimport utility.
Reason: null

There might be a problem during java artifacts creation: for example a name
conflict in generated classes.
To detect the problem see also the error messages in the output window.
You may be able to fix the problem in the WSDL Customization dialog (Edit Web
Serive Attributes action)
or by manual editing of the local wsdl or schema files, using the JAXB customization
(local wsdl and schema files are located in the xml-resources directory).

After you fix the problem you need to refresh the client from the Web Service
References node.

OK

i then click on the OK button, try to refresh the click and select the checkbox
to replace wsdl, and i get same error dialog as shown above, along with this in
the output window:

wsimport-client-clean-SimpleService:
Deleting directory
/wsittutorial/examples/wstx/basicWSTX/SampleServiceClient/build/generated/wsimport/client/wstx/sample/client
init:
wsimport-init:
wsimport-client-check-SimpleService:
wsimport-client-SimpleService:
Consider using <depends>/<produces> so that wsimport won't do unnecessary
compilation
parsing WSDL...

generating code...
BUILD SUCCESSFUL (total time: 4 seconds)

and this in the Retriever Output window:

Apr 3, 2007 7:37:15 PM : Retrieving Location:
[http://localhost:8080/SampleService-war/SimpleService?wsdl](http://localhost:8080/SampleService-war/SimpleService?wsdl)
Retrieved : [http://localhost:8080/SampleService-war/SimpleService?wsdl](http://localhost:8080/SampleService-war/SimpleService?wsdl)
Saved at:
/files/share/workspaces/m4_wsit_tutorial/wsittutorial/examples/wstx/basicWSTX/SampleServiceClient/src/conf/xml-resources/web-service-references/SimpleService/wsdl/localhost_8080/SampleService-war/SimpleService.wsdl
Retrieving Location: [http://localhost:8080/SampleService-war/SimpleService?xsd=1](http://localhost:8080/SampleService-war/SimpleService?xsd=1)
Found in document:
[http://localhost:8080/SampleService-war/SimpleService?wsdl](http://localhost:8080/SampleService-war/SimpleService?wsdl)
Retrieved : [http://localhost:8080/SampleService-war/SimpleService?xsd=1](http://localhost:8080/SampleService-war/SimpleService?xsd=1)
Saved at:
/files/share/workspaces/m4_wsit_tutorial/wsittutorial/examples/wstx/basicWSTX/SampleServiceClient/src/conf/xml-resources/web-service-references/SimpleService/wsdl/localhost_8080/SampleService-war/SimpleService.xsd_1.xsd

so i'm blocked on step 9.
#### Environment
Operating System: All
Platform: Sun
#### Affected Versions
[current]
